### PR TITLE
Schema func

### PR DESCRIPTION
**Pull request by smacker**
Fix: #50 

based on #53 

- change prop
- add test
- fix bug found by test

### CODE REVIEW

File: src/components/Node.js [hunk 1]
```diff
@@ -122,15 +129,7 @@ class PureNode extends PureComponent {
...
-  schema: PropTypes.arrayOf(
-    PropTypes.shape({
-      name: PropTypes.string.isRequired,
-      attr: PropTypes.func.isRequired,
-      type: PropTypes.oneOf(['array', 'object', 'location']),
-      label: PropTypes.string,
-      showEmpty: PropTypes.bool
-    })
-  ).isRequired,
+  schema: PropTypes.func.isRequired,
```
Comment (carlosms) [inline src/components/Node.js]: Maybe we could put the expected return type as a comment, copying the deleted `PropTypes.arrayOf & shape`.

File: src/components/Node.js [hunk 2]
```diff
@@ -129,6 +129,19 @@ class PureNode extends PureComponent {
 PureNode.propTypes = {
   id: PropTypes.number.isRequired,
   node: PropTypes.object.isRequired,
+  /*
+  return value of the schema function must be in the share:
```
Comment (carlosms) [inline src/components/Node.js]: typo, share -> shape?

**Comment by smacker** [comment]
tests are failing after rebase on master. I'm on it.

**Comment by smacker** [comment]
fixed

**Pull request merged**